File: src/Safir.Agent/Services/IFileWatcher.cs
using System;
using System.IO;

namespace Safir.Agent.Services
{
    public interface IFileWatcher
    {
        IObservable<FileSystemEventArgs> Created { get; }
        
        IObservable<FileSystemEventArgs> Changed { get; }
        
        IObservable<FileSystemEventArgs> Deleted { get; }
        
        IObservable<RenamedEventArgs> Renamed { get; }
        
        IObservable<ErrorEventArgs> Error { get; }
    }
}

File: src/agent/src/Safir.Agent/Services/IFileWatcher.fs
namespace Safir.Agent.Services

open System
open System.IO

type IFileWatcher =
    abstract member Created: IObservable<FileSystemEventArgs>
    abstract member Changed: IObservable<FileSystemEventArgs>
    abstract member Deleted: IObservable<FileSystemEventArgs>
    abstract member Renamed: IObservable<RenamedEventArgs>
    abstract member Error: IObservable<ErrorEventArgs>
